api/v1/categories/PkpCategoryCategoryController.php

```diff
@@ -0,0 +1,409 @@
+
+        $data = Repo::category()->getSchemaMap()->mapMany($categories)->values();
+
+        return response()->json($data, Response::HTTP_OK);
+    }
+
+    /**
+     * Get the category form component.
+     * Pass `categoryId` as a query parameter to load the form for editing an existing category.
+     */
+    public function getCategoryFormComponent(Request $illuminateRequest): JsonResponse
+    {
+        $categoryId = $illuminateRequest->input('categoryId');
+        $request = Application::get()->getRequest();
+        $context = $request->getContext();
+
+        if ($categoryId) {
+            $category = Repo::category()->get($categoryId, $context->getId());
+
+            if (!$category) {
+                return response()->json(__('api.404.resourceNotFound'), Response::HTTP_NOT_FOUND);
+            }
+        }
+
+        $locales = $context->getSupportedFormLocaleNames();
+        $locales = array_map(fn (string $locale, string $name) => ['key' => $locale, 'label' => $name], array_keys($locales), $locales);
+        $publicFileManager = new PublicFileManager();
+        $baseUrl = $request->getBaseUrl() . '/' . $publicFileManager->getContextFilesPath($context->getId());
+        $temporaryFileApiUrl = $request->getDispatcher()->url($request, Application::ROUTE_API, $context->getPath(), 'temporaryFiles');
+
+        $categoriesApiUrl = $request->getDispatcher()->url(
+            Application::get()->getRequest(),
+            Application::ROUTE_API,
+            $context->getPath(),
+            'categories'
+        );
+
+        $form = new CategoryForm($categoriesApiUrl, $locales, $baseUrl, $temporaryFileApiUrl, $category ?: null);
+        return response()->json($form->getConfig(), Response::HTTP_OK);
+    }
+
+    /**
+     * Delete a category by ID.
+     */
+    public function delete(Request $illuminateRequest): JsonResponse
+    {
+        $categoryId = $illuminateRequest->route('categoryId');
+        $context = $this->getRequest()->getContext();
+
+        $category = Repo::category()->get($categoryId, $context->getId());
+        if (!$category) {
+            return response()->json(__('api.404.resourceNotFound'), Response::HTTP_NOT_FOUND);
+        }
+
+        $image = $category->getImage();
+        Repo::category()->delete($category);
+
+        if ($image) {
+            $publicFileManager = new PublicFileManager();
+            $publicFileManager->removeContextFile($category->getContextId(), $image['uploadName']);
+            $publicFileManager->removeContextFile($category->getContextId(), $image['thumbnailName']);
+        }
+
+        return response()->json([], Response::HTTP_OK);
+    }
+
+    /**
+     * Generate the thumbnail image when creating a category.
+     *
+     * @return array - assoc array with thumbnail details
+     * Example:
+     * ```
+     * [
+     *  'thumbnailName' => (string),
+     *  'thumbnailWidth' => (int),
+     *  'thumbnailHeight' => (int),
+     * ]
+     * ```
```

Comment:
I think we can use the array shape syntax (https://phpstan.org/writing-php-code/phpdoc-types#array-shapes)